backend/wmg/data/query.py

```diff
@@ -153,7 +155,25 @@ def _query(
 
         tiledb_dims_query = tuple(tiledb_dims_query)
 
-        query_result_df = cube.query(cond=query_cond or None, use_arrow=True).df[tiledb_dims_query]
+        if isinstance(criteria, FmgQueryCriteria):
+            # fmg queries retain existing behavior of pulling all dims and attrs
+            query_result_df = cube.query(cond=query_cond or None, use_arrow=True).df[tiledb_dims_query]
```

Comment:
not entirely sure if this is right, but when passing in the "no compare param" logic, all of the tests fail with this:
```
E   tiledb.cc.TileDBError: Selected dimension does not exist: 'gene_ontology_term_id'
```

so i'm guessing that we want to retain the existing behavior for FMG queries, and only do the optimization for WMG